### PR TITLE
Move federated_share_added into a typed event

### PR DESCRIPTION
**Pull request by MorrisJobke**
This only needs documentation. Should I just add it to https://docs.nextcloud.com/server/19/developer_manual/app/events.html#available-events ?

Also should those events be in the apps or in `OCP`?

* [x] add to documentation once naming is decided https://docs.nextcloud.com/server/19/developer_manual/app/events.html#available-events
* [x] Add to #20953 if the old hook is dropped
* [x] ~~Add to #14552 if there was a fallback added for the old hook that needs to be removed with the removal of hooks~~

### CODE REVIEW

**Comment by ChristophWurst** [comment]
> Also should those events be in the apps or in `OCP`?

How clean are we with DAV related code? If other abstractions are in OCP I would vote for that. If there is more stuff that DAV apps use from the dav app directly then we could also just follow that pattern and leave in in dav.

**Comment by rullzer** [comment]
Well we also have events emitted by the files app and by the sharing app. The issue is you get this weird mix if we move them to OCP.

I'd say with the new events it is fine to have them in the apps themselves. That is where they belong.
Also it should all be fine since you use the ::class to get the event so even if you don'thave the app (not possible in this case) it should still work.

File: apps/dav/lib/Events/SabrePluginAuthInitEvent.php [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+ *
+ */
+
+namespace OCA\DAV\Events;
+
+use OCP\EventDispatcher\Event;
+use Sabre\DAV\Server;
+
+/**
+ * @since 20.0.0
```
Comment (ChristophWurst) [inline apps/dav/lib/Events/SabrePluginAuthInitEvent.php:33]: mind adding a little on *when* this is emitted?

**Review by ChristophWurst — APPROVED** [review]

File: apps/dav/lib/Events/SabrePluginAuthInitEvent.php (hunk 1 above)
Comment (MorrisJobke) [reply apps/dav/lib/Events/SabrePluginAuthInitEvent.php:36]: Done.

**Review by ChristophWurst — APPROVED** [review]

**Review by rullzer — APPROVED** [review]
Didn't test but looks sane!

**Comment by MorrisJobke** [comment]
Time to squash the fixups. Ready to get in :)

**Comment by MorrisJobke** [comment]
Failures are the ones that fail regularly but not always. Thus merging and documenting this.

**Comment by MorrisJobke** [comment]
> ####
> * build/integration/features/auth.feature:70

This was actually a bug introduced here. Let me debug it.

**Pull request merged**